src/dotnet/APIView/APIViewWeb/Languages/LanguageProcessor.cs

```diff
@@ -42,34 +44,67 @@ public override async Task<CodeFile> GetCodeFileAsync(string originalName, Strea
+            var processStartInfo = new ProcessStartInfo(ProcessName, arguments);
+            string processErrors = String.Empty;
+
+            processStartInfo.WorkingDirectory = workingDirectory;
+            processStartInfo.RedirectStandardError = true;
+            processStartInfo.RedirectStandardOutput = true;
+            processStartInfo.UseShellExecute = false;
+            processStartInfo.CreateNoWindow = true;
+
+            var output = new StringBuilder();
+            var error = new StringBuilder();
+
+            using (var process = new Process())
             {
-                var arguments = GetProcessorArguments(originalName, tempDirectory, jsonFilePath);
-                var processStartInfo = new ProcessStartInfo(ProcessName, arguments);
-                string processErrors = String.Empty;
+                process.StartInfo = processStartInfo;
+                process.OutputDataReceived += (sender, args) =>
+                {
+                    if (args.Data != null)
+                        output.AppendLine(args.Data);
+                };
 
-                processStartInfo.WorkingDirectory = tempDirectory;
-                processStartInfo.RedirectStandardError = true;
-                processStartInfo.RedirectStandardOutput = true;
+                process.ErrorDataReceived += (sender, args) =>
+                {
+                    if (args.Data != null)
+                        error.AppendLine(args.Data);
+                };
+                process.Start();
+                process.BeginOutputReadLine();
+                process.BeginErrorReadLine();
+                process.WaitForExit();
 
-                using (var process = Process.Start(processStartInfo))
+                if (process.ExitCode != 0)
                 {
-                    process.WaitForExit();
-                    if (process.ExitCode != 0)
-                    {
-                        processErrors = "Processor failed: " + Environment.NewLine +
-                            "stdout: " + Environment.NewLine +
-                            process.StandardOutput.ReadToEnd() + Environment.NewLine +
-                            "stderr: " + Environment.NewLine +
-                            process.StandardError.ReadToEnd() + Environment.NewLine;
-                        throw new InvalidOperationException(processErrors);
-                    }
+                    processErrors = "Processor failed: " + Environment.NewLine +
+                        "stdout: " + Environment.NewLine +
+                        output + Environment.NewLine +
+                        "stderr: " + Environment.NewLine +
+                        error + Environment.NewLine;
+                    throw new InvalidOperationException(processErrors);
                 }
+            }
+            return processErrors;
+        }
+
+        public async Task<CodeFile> RunParserProcess(string originalName, string tempDirectory, string jsonPath, string language)
```

Comment:
You don't need language parameter. You can use `Name` which is the language name of the service class.